Base 64 Encoding Of Data

AlAtEX opened the pull request:
This uses the android.util.Base64 method of encoding to base 64, and will modify the data before it is saved/sent.

The server-side portion of this can easily be implemented by taking the portion where the server loads the file and decoding it, using this method:

```java
byte[] data = Base64.decode(encoded_data, Base64.DEFAULT);
String text = new String(data, Charset.forName("UTF-8"));
```

AlAtEX committed:
Merge pull request #1 from LakeEffectRobotics/master

Rebase

AlAtEX committed:
Encode sent data to base 64 using android.util.Base64.

AlAtEX committed:
Merge branch 'master' of https://github.com/AlAtEX/LakeEffectScoutingApp

ajayyy commented on app/src/main/java/ca/lakeeffect/scoutingapp/MainActivity.java:
```diff
@@ -447,7 +450,13 @@ public void run() {
-        return out;
+        byte[] data2 = out[0].getBytes(Charset.forName("UTF-8"));
+        byte[] labels2 = out[1].getBytes(Charset.forName("UTF-8"));
+        String data_base64 = Base64.encodeToString(data2, Base64.DEFAULT);
```
camelCasePlease :)

`dataBase64`

ajayyy commented on app/src/main/java/ca/lakeeffect/scoutingapp/MainActivity.java:
```diff
@@ -447,7 +450,13 @@ public void run() {
         String[] out = {data.toString(), labels.toString()};
-        return out;
+        byte[] data2 = out[0].getBytes(Charset.forName("UTF-8"));
+        byte[] labels2 = out[1].getBytes(Charset.forName("UTF-8"));
```
I think this would work better if you worked with the data and labels variable directly, removing the out variable entirely.

Also, I suggest renaming `data2` and `labels2` to something like `dataBytes`

ajayyy commented on app/src/main/java/ca/lakeeffect/scoutingapp/MainActivity.java:
```diff
@@ -54,6 +56,7 @@
 import java.util.List;
 import java.util.UUID;
 
+
```
A nitpick, but this enter is unnessesary :)

ajayyy reviewed (changes requested):
Nice job! Thanks for working through this on your own.

I don't think it's a good idea to merge this until we have all the code setup on the serverside as well to make sure both apps always work together on the same release and so we can test the code though.

Make sure to make a new branch **without** these commits if working on another feature at another time.

I also made some change suggestions.

Again, thanks for working this out on your own, obviously code style is a personal thing so it is hard to guess the style me and Greg like to use, but I still want to stick with one consistent style throughout the project.

Another thing to keep in mind is that we must make sure how this data is being used, because we also save "pending data" onto the phone as a string, so if this is now in base 64, we need to add the decoder when loading that back up.

AlAtEX committed:
Revised Code of #176

ajayyy committed:
Setup base64 conversion to work for event data as well plus fixes

Saves local files properly still, converting back to a normal string
Converts the full data sent to base64 as well before sending.

ajayyy committed:
Decodes schedules from server in Base64

ajayyy committed:
Fixes base 64 encoding issues

Bytes were being sent without being converted to a string, things weren't being encoded. Added a nice function to quickly convert to base 64.
